src/Compilers/CSharp/Portable/Symbols/Source/LambdaSymbol.cs

```diff
@@ -310,11 +260,36 @@ public override bool IsExtensionMethod
             get { return false; }
         }
 
-        public override bool HidesBaseMethodsByName
+        internal SyntaxNode Syntax => syntaxReferenceOpt.GetSyntax();
+
+        internal override OneOrMany<SyntaxList<AttributeListSyntax>> GetAttributeDeclarations()
         {
-            get { return false; }
+            // PROTOTYPE: Move AttributeLists to LambdaExpressionSyntax base class.
+            var attributeLists = Syntax switch
+            {
+                SimpleLambdaExpressionSyntax syntax => syntax.AttributeLists,
+                ParenthesizedLambdaExpressionSyntax syntax => syntax.AttributeLists,
+                _ => default
+            };
+            return OneOrMany.Create(attributeLists);
+        }
+
+        internal void GetDeclarationDiagnostics(BindingDiagnosticBag addTo)
```

Comment:
I'm not certain why we need both `GetDeclarationDiagnostics` and `AddDeclarationDiagnostics`?

Reply:
`AddDeclarationDiagnostics()` is an override of a method called in several places (such as from `SourceParameterSymbol`) where declaration diagnostics may be calculated. `GetDeclarationDiagnostics()` provides access to the declaration diagnostics which is necessary from `UnboundLambdaState.ReallyBind()` to ensure the diagnostics that are calculated are actually reported.

---
In reply to: [605833109](https://github.com/dotnet/roslyn/pull/52178#discussion_r605833109) [](ancestors = 605833109)